ksim/env/builders/loggers.py

```diff
@@ -0,0 +1,136 @@
+"""Defines a standard interface for logging metrics."""
+
+from abc import ABC, abstractmethod
+from typing import Generic, NamedTuple, Optional, TypeVar
+
+import attrs
+import jax.numpy as jnp
+import xax
+from jaxtyping import Array, PyTree
+
+from ksim.utils.data import BuilderData
+
+
+class TrajectoryData(NamedTuple):
+    """Simple trajectory data structure for logging.
+
+    TODO: This class should really be PPOBatch, but that leads to circular imports.
+    So we should have a base "Batch" type in a seperate file that both PPO and
+    logging can import from.
+    """
+
+    observations: PyTree
+    next_observations: PyTree
+    actions: Array
+    rewards: Array
+    done: Array
+    action_log_probs: Array
+
+
+@attrs.define(frozen=True, kw_only=True)
+class LoggingData:
+    trajectory: Optional[TrajectoryData] = None
```

Comment:
prefer python 3.11 types, i.e.,

```suggestion
    trajectory: TrajectoryData | None = None
```

unless there's a reason not to